server/controllers/boards.js

```diff
@@ -41,6 +42,57 @@ exports.getBoard = asyncHandler(async (req, res) => {
   }
 })
 
+exports.getBoardFull = asyncHandler(async (req, res) => {
+  const boardId  = req.params.boardId;
+  
+
+  const board = await Board.findById(boardId);
+  /*const columns = await Promise.all(board.columns.map(async (id) => {
+    let column = {};
+    const doc = await Column.findById(id);
+    column[doc._id] = doc;
+    return column;
+  }));
+  */
+ const columns = await getColumsAsObject(board);
+
+
+  const cards = await getCardsColums(board, columns);
+  /*await Promise.all (columns.map( async column => {
+    console.log("cards");
+
+    cards = await Promise.all( (column.cards.map(async (id) => {
+    console.log("col: " + id);
+
+      const doc = await Card.findById(id);
+      console.log("doc: " + doc)
+      cards[doc._id] = doc;
+
+    })));
+
+    console.log("card" + cards);
+
+  }));
+  */
+
+
+
+  if (board && columns && cards) {
+    res.status(200).json({ board, columns, cards });
+
+  } else {
+    const error = new Error(`Could not find board ${boardId}`);
+    res.status(404).json({
+      error
+    })
```

Comment:
You could set the status of the response then throw an error and it express-async-handler will respond with a error json response

Reply:
```
  const error = new Error(`Could not find card ${cardId}`);
  res.status(404).error({
    error
  })
}
```
Would this be correct?